custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,421 @@
+            
+            # Only match first entry
+            entry_property_value = entry.get(dict_field, None)
+            if entry_property_value is not None:
+                entry_id = entry.get('id')
+                anilist_id = entry.get('anilist_id', None)
+                entry_name = entry.get('name')
+                
+                logger.info(f"Matched entry: ID: '{entry_id}', anilist_id: '{anilist_id}', name: '{entry_name}', english_name: '{entry.get('english_name')}'")
+                if entry.get("flags").get("unverified"):
+                    logger.warning(f"This entry '{entry_id}' is unverified, subtitles might be incomplete or have quality issues!")
+                
+                break
+            
+        if entry_property_value is None:
+            logger.error('Matched no entries.')
+            return None
+        
+        # Get a list of subtitles for entry
+        retry_count = 0
+        while True:
+            episode_number = video.episode
+            url = f"entries/{entry_id}/files?episode={episode_number}"
+            data = self._get_jimaku_response(url)
+            
+            # Edge case: When dealing with a cour, episodes could be uploaded with their episode numbers having an offset applied
+            if not data and self.video_episode_number_override and retry_count < 1:
+                episode_number = self.video_episode_number_override
+                logger.warning(f"Found no subtitles for {episode_number}, but will retry with offset-adjusted episode number {self.video_episode_number_override}.")
+                retry_count += 1
+            elif not data:
+                return None
+            else:
+                break
+        
+        # Filter subtitles
+        list_of_subtitles = []
+        
+        archive_formats_blacklist = (".7z",) # Unhandled format
+        if not self.enable_archives:
+            disabled_archives = (".zip", ".rar")
+            
+            # Handle shows that only have archives uploaded
+            filter = [item for item in data if not item['name'].endswith(disabled_archives)]
+            if len(filter) == 0:
+                logger.warning("Archives are disabled, but only archived subtitles have been returned. Will therefore download anyway.")
+            else:
+                archive_formats_blacklist += disabled_archives
+
+        # Order subtitles to maximize quality
+        # Certain subtitle sources, such as Netflix/WebRip usually yield superior quality over others
+        sorted_data = sorted(data, key=self._subtitle_sorting_key)
+
+        for item in sorted_data:
+            subtitle_filename = item.get('name')
+            subtitle_filesize = item.get('size')
+            subtitle_url = item.get('url')
+            
+            if not self.enable_ai_subs:
+                #ToDo: Implement
```

Comment:
Implement what?

Reply:
Oversight, removed in [`db5f56c`](https://github.com/morpheus65535/bazarr/pull/2505/commits/db5f56c2ffde69f2bde023fd79bbbdb0df575a20).